### PR TITLE
added ci job for open api gen

### PR DESCRIPTION
**Pull request by EraKin575**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: #5679 

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by Shelley-BaoYue — COMMENTED** [review]
Hi, @EraKin575  Thanks for your contributions!  

If you still work in progress, you can add a [wip] at title. 
I think put this job in https://github.com/kubeedge/kubeedge/blob/master/.github/workflows/main.yaml is more suitable. : )

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -392,6 +392,21 @@ jobs:
           path: |
             /tmp/cloudcore.log
             /tmp/edgecore.log
+  check-api-docs:
```
Comment (Shelley-BaoYue) [inline .github/workflows/main.yaml:395]: I have a new idea:  we don't need add a new job here. We can add a verify under `make verify`  in https://github.com/kubeedge/kubeedge/blob/master/Makefile#L67.  This job https://github.com/kubeedge/kubeedge/blob/master/.github/workflows/main.yaml#L25 will run `make verify`.

File: tests/scripts/check-api-docs.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
```
Comment (Shelley-BaoYue) [inline tests/scripts/check-api-docs.sh:1]: Would it be better name this script `verify-apidoc.sh` to consistent with other verify scripts? And as said above, just add a `verify-apidoc` under make verify https://github.com/kubeedge/kubeedge/blob/master/Makefile#L67 is enough.

Comment (EraKin575) [reply tests/scripts/check-api-docs.sh:1]: thanks for the review! I will make the changes

Comment (EraKin575) [reply tests/scripts/check-api-docs.sh]: Can I just implement the changes to the generated file if it needs to be generated after verifying or just leave it at verification?

File: Makefile [hunk 3]
```diff
@@ -91,9 +91,12 @@ verify-codegen:
 .PHONY: verify-vendor-licenses
 verify-vendor-licenses:
 	hack/verify-vendor-licenses.sh
-.PHONY: verify-crds
-verify-crds:
-	hack/verify-crds.sh
+# .PHONY: verify-crds
```
Comment (Shelley-BaoYue) [inline Makefile:94]: ditto

File: hack/verify-apidoc.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The KubeEdge Authors.
```
Comment (Shelley-BaoYue) [inline hack/verify-apidoc.sh:3]:
```suggestion
# Copyright 2024 The KubeEdge Authors.
```

**Review by Shelley-BaoYue — COMMENTED** [review]
why apidoc/tools/update-swagger-docs.sh updated? 
image

File: Makefile (hunk 3 above)
Comment (EraKin575) [reply Makefile:94]: apologies, verify-crds is not working properly, I had to remove it to run make-verify which was after it. I forgot to put it back

**Comment by EraKin575** [comment]
closing this pr

**Comment by EraKin575** [comment]
@Shelley-BaoYue Closing this PR, and opening another one with clean commit history

**Pull request closed without merging**